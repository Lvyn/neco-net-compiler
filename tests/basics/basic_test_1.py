from snakes.nets import *

net = PetriNet('Net')
net.processes = []

testplace = Place('testplace', [dot], tBlackToken)
testplace.one_safe = True
testplace.process_name = None
testplace.flow_control = False

s1 = Place('s1', [dot], tBlackToken)
s1.one_safe = True
s1.process_name = 'a'
s1.flow_control = False

s2 = Place('s2', [], tBlackToken)
s2.one_safe = True
s2.process_name = 'a'
s2.flow_control = False

net.add_place(s1)
net.add_place(s2)
net.add_place(testplace)

transition = Transition('t', Expression('True'))
net.add_transition(transition)

net.add_input('s1', 't', Value(dot))
net.add_input('testplace', 't', Test(Value(dot)))
net.add_output('s2', 't', Value(dot))
