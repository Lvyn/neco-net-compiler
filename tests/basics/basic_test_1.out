check - markings
begin marking
s1 - dot
s2 -
testplace - dot
end marking
begin marking
s1 -
s2 - dot
testplace - dot
end marking
